families/supplychain/python/sawtooth_supplychain/rest_api/route_handlers.py

```diff
@@ -0,0 +1,795 @@
+            LOGGER.debug("Could not connect to database.")
+            raise errors.DatabaseConnectionError()
+
+        cur = conn.cursor()
+
+        # Execute the query
+        try:
+            cur.execute(query, query_tuple)
+        except:
+            LOGGER.debug("Could not execute query: %s", query)
+            raise errors.UnknownDatabaseError()
+
+        # Fetch the rows
+        rows = cur.fetchall()
+
+        # Close DB connection
+        try:
+            self._db_close(conn, cur)
+        except:
+            LOGGER.debug("Could not close database connection.")
+            raise errors.UnknownDatabaseError()
+
+        fields = ('identifier', 'name')
+        data = self._make_dict(fields, rows)
+
+        # Return the data
+        return self._wrap_response(
+            request,
+            data=data,
+            metadata="")
+
+    async def fetch_agent(self, request):
+        """Fetches a specific agent, by identifier.
+
+        Request:
+            query:
+                - agent_id: the identifier of the agent to fetch
+
+        Response:
+            data: JSON response with Agent data
+        """
+
+        agent_id = request.match_info.get('agent_id', '')
+
+        # Create query
+        query = "select identifier, name from agent where identifier = %s;"
+
+        # Get a connection and cursor
+        try:
+            conn = self._db_connect()
+        except psycopg2.OperationalError:
+            LOGGER.debug("Could not connect to database.")
+            raise errors.DatabaseConnectionError()
+
+        cur = conn.cursor()
+
+        # Execute the query
+        try:
+            cur.execute(query, (agent_id,))
+        except:
+            LOGGER.debug("Could not execute query: %s", query)
+            raise errors.UnknownDatabaseError()
+
+        # Fetch the rows
+        rows = cur.fetchall()
+        if len(rows) > 1:
+            LOGGER.debug("Too many rows returned.")
+            errors.UnknownDatabaseError()
+
+            # Close DB connection
+            try:
+                self._db_close(conn, cur)
+            except:
+                LOGGER.debug("Could not close database connection.")
+                raise errors.UnknownDatabaseError()
+
+        elif len(rows) == 1:
+            # Close DB connection
+            try:
+                self._db_close(conn, cur)
+            except:
+                LOGGER.debug("Could not close database connection.")
+                raise errors.UnknownDatabaseError()
+
+            # Return the data
+            fields = ('identifier', 'name')
+            data = self._make_dict(fields, rows).pop()
+
+            return self._wrap_response(
+                request,
+                data=data,
+                metadata="")
+
+        else:
+            # Close DB connection
+            try:
+                self._db_close(conn, cur)
+            except:
+                LOGGER.debug("Could not close database connection.")
```

Comment:
This should be logged as an error, or exception - it is a pretty bad server error that an admin should be aware of.  Correct this at every use of `UnknownDatabaseError`